go-controller/pkg/kube/kube.go

```diff
@@ -15,7 +15,7 @@ import (
 // Interface represents the exported methods for dealing with getting/setting
 // kubernetes resources
 type Interface interface {
-	SetAnnotationOnPod(pod *kapi.Pod, key, value string) error
+	SetAnnotationOnPod(namespace, name string, key, value string) error
```

Comment:
please change to 
`SetAnnotationOnPod(namespace, name, key, value string) error`